android/java/org/chromium/chrome/browser/crypto_wallet/activities/AssetDetailActivity.java

```diff
@@ -93,12 +95,22 @@
     private boolean mNativeInitialized;
     private boolean mShouldShowDialog;
     private WalletModel mWalletModel;
+    private NetworkInfo mAssetNetwork;
 
     @Override
     protected void triggerLayoutInflation() {
         setContentView(R.layout.activity_asset_detail);
+        try {
+            BraveActivity activity = BraveActivity.getBraveActivity();
+            mWalletModel = activity.getWalletModel();
+            if (mWalletModel == null) {
+                finish();
+                return;
+            }
+        } catch (ActivityNotFoundException ignored) {
```

Comment:
This is very unlikely to happen, right?

Shall we consider calling `finish()` even inside the catch clause?

Reply:
let's just keep as is for now. We have an issue to go away from the activity, we will figure it out eventually